lib/handler/cache-handler.js

```diff
@@ -0,0 +1,351 @@
+'use strict'
+
+const util = require('../core/util')
+const DecoratorHandler = require('../handler/decorator-handler')
+const { parseCacheControlHeader, parseVaryHeader, UNSAFE_METHODS } = require('../util/cache')
+
+/**
+ * Writes a response to a CacheStore and then passes it on to the next handler
+ */
+class CacheHandler extends DecoratorHandler {
+  /**
+   * @type {import('../../types/cache-interceptor.d.ts').default.CacheOptions} | null
+   */
```

Comment:
Still i hate that we pass around the opts. but here the jsdoc is wrong.


```suggestion
   * @type {import('../../types/cache-interceptor.d.ts').default.CacheOptions | null}
```